### PR TITLE
feat: Simple bot protection that should elimiminate most of the spam on our forms

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Add simple protection against non js and jsdom bots

Idea is to add hidden js generated field on submit, and check that matchMedia is working.

## Steps for reproduction

Go here
https://saas-form-rdlrc.wstd.work/

then check via curl

```bash
curl 'https://saas-form-rdlrc.wstd.work/?index=&_data=routes%2F_index' \
  -H 'content-type: application/x-www-form-urlencoded;charset=UTF-8' \
  -H 'referer: https://saas-form-rdlrc.wstd.work/' \
  --data-raw 'ws--form-id=2TIgyB2vQViDMitiYiGSK&name=wewe&email=wewe&ws--form-bot=18f72a6'

curl 'https://saas-form-rdlrc.wstd.work/?index=&_data=routes%2F_index' \
  -H 'content-type: application/x-www-form-urlencoded;charset=UTF-8' \
  -H 'referer: https://saas-form-rdlrc.wstd.work/' \
  --data-raw 'ws--form-id=2TIgyB2vQViDMitiYiGSK&name=wewe&email=wewe'

curl 'https://saas-form-rdlrc.wstd.work/?index=&_data=routes%2F_index' \
  -H 'content-type: application/x-www-form-urlencoded;charset=UTF-8' \
  -H 'referer: https://saas-form-rdlrc.wstd.work/' \
  --data-raw 'ws--form-id=2TIgyB2vQViDMitiYiGSK&name=wewe&email=wewe&ws--form-bot=jsdom'
```
## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/cli/templates/defaults/app/route-templates/html.tsx [hunk 1]
```diff
@@ -240,66 +244,83 @@ const getMethod = (value: string | undefined) => {
...
+      throw new Error("Form bot field not found");
+    }
 
-  // wrapped in try/catch just in cases new URL() throws
-  // (should not happen)
-  let pageUrl: URL;
-  try {
-    pageUrl = new URL(request.url);
+    const submitTime = parseInt(formBotValue, 16);
+    // 5 minutes
```
Comment (kof) [inline packages/cli/templates/defaults/app/route-templates/html.tsx:269]: whats the idea about 5 minutes, please write full sentences, I am guessing its the time between user clicking the submit button and server receiving the data?

File: packages/sdk-components-react-remix/src/webhook-form.tsx [hunk 2]
```diff
@@ -55,8 +100,28 @@ export const WebhookForm = forwardRef<
...
+     * Add hidden field generated using js with simple jsdom detector.
+     * This is used to protect form submission against very simple bots.
+     */
+    const handleSubmitAndAddHiddenJsField = (
+      event: React.FormEvent<HTMLFormElement>
+    ) => {
+      const hiddenInput = document.createElement("input");
+      hiddenInput.type = "hidden";
+      hiddenInput.name = formBotFieldName;
+      hiddenInput.value = isJSDom() ? "jsdom" : Date.now().toString(16);
```
Comment (kof) [inline packages/sdk-components-react-remix/src/webhook-form.tsx:113]: I couldn't find where we treat this jsdom value

**Review by kof — APPROVED** [review]

File: fixtures/webstudio-cloudflare-template/app/routes/_index.tsx [hunk 3]
```diff
@@ -240,66 +244,83 @@ const getMethod = (value: string | undefined) => {
...
+    if (formBotValue == null || typeof formBotValue !== "string") {
+      throw new Error("Form bot field not found");
+    }
 
-  // wrapped in try/catch just in cases new URL() throws
-  // (should not happen)
-  let pageUrl: URL;
-  try {
-    pageUrl = new URL(request.url);
+    const submitTime = parseInt(formBotValue, 16);
```
Comment (TrySound) [inline fixtures/webstudio-cloudflare-template/app/routes/_index.tsx:268]: Number.parseInt please

Comment (kof) [reply fixtures/webstudio-cloudflare-template/app/routes/_index.tsx:268]: or even Number.parseFloat please haha

Comment (TrySound) [reply fixtures/webstudio-cloudflare-template/app/routes/_index.tsx:268]: parseFloat, no
need to parse hex here

File: fixtures/webstudio-cloudflare-template/app/routes/_index.tsx [hunk 4]
```diff
@@ -240,66 +244,83 @@ const getMethod = (value: string | undefined) => {
...
-  const formId = getFormId(formData);
-  if (formId === undefined) {
-    // We're throwing rather than returning { success: false }
-    // because this isn't supposed to happen normally: bug or malicious user
-    throw json("Form not found", { status: 404 });
-  }
+    const formId = formData.get(formIdFieldName);
 
-  const formProperties = formsProperties.get(formId);
+    if (formId == null || typeof formId !== "string") {
```
[the remark this answers is not in the document]
Comment (istarkov) [reply fixtures/webstudio-cloudflare-template/app/routes/_index.tsx:258]: Yep useles.

File: packages/cli/templates/defaults/app/route-templates/html.tsx [hunk 5]
```diff
@@ -240,66 +244,83 @@ const getMethod = (value: string | undefined) => {
...
+    if (
```
Comment (istarkov) [inline packages/cli/templates/defaults/app/route-templates/html.tsx:270]: Commented

File: packages/sdk-components-react-remix/src/webhook-form.tsx (hunk 2 above)
Comment (kof) [reply packages/sdk-components-react-remix/src/webhook-form.tsx:114]: Ok, wouldn't guess it and would assume the value is used somehow, needs at least a comment

**Pull request merged**